interp/builtin.go

```diff
@@ -538,6 +544,14 @@ func (r *Runner) builtinCode(ctx context.Context, pos syntax.Pos, name string, a
 			}
 		}
 
+		if prompt.provided {
+			if prompt.message == "" {
```

Comment:
you could make this check inside `case "-p"`, then you don't need `provided` at all - you'd just do `message != ""`.

and you could also just do `message := ""`; I don't think the struct is necessary

Reply:
Ok gotcha, making the change 👍